src/trig.c

```diff
@@ -6,167 +6,28 @@
-    0xeff5, 0xf018, 0xf03b, 0xf05e, 0xf080, 0xf0a3, 0xf0c5, 0xf0e7, 0xf109,
-    0xf12b, 0xf14c, 0xf16e, 0xf18f, 0xf1b1, 0xf1d2, 0xf1f3, 0xf213, 0xf234,
-    0xf254, 0xf275, 0xf295, 0xf2b5, 0xf2d5, 0xf2f5, 0xf314, 0xf334, 0xf353,
-    0xf372, 0xf391, 0xf3b0, 0xf3cf, 0xf3ed, 0xf40c, 0xf42a, 0xf448, 0xf466,
-    0xf484, 0xf4a2, 0xf4bf, 0xf4dd, 0xf4fa, 0xf517, 0xf534, 0xf551, 0xf56e,
-    0xf58a, 0xf5a6, 0xf5c3, 0xf5df, 0xf5fb, 0xf616, 0xf632, 0xf64e, 0xf669,
-    0xf684, 0xf69f, 0xf6ba, 0xf6d5, 0xf6ef, 0xf70a, 0xf724, 0xf73e, 0xf758,
-    0xf772, 0xf78c, 0xf7a5, 0xf7bf, 0xf7d8, 0xf7f1, 0xf80a, 0xf823, 0xf83b,
-    0xf854, 0xf86c, 0xf885, 0xf89d, 0xf8b4, 0xf8cc, 0xf8e4, 0xf8fb, 0xf913,
-    0xf92a, 0xf941, 0xf958, 0xf96e, 0xf985, 0xf99b, 0xf9b2, 0xf9c8, 0xf9de,
-    0xf9f3, 0xfa09, 0xfa1f, 0xfa34, 0xfa49, 0xfa5e, 0xfa73, 0xfa88, 0xfa9c,
-    0xfab1, 0xfac5, 0xfad9, 0xfaed, 0xfb01, 0xfb15, 0xfb28, 0xfb3c, 0xfb4f,
-    0xfb62, 0xfb75, 0xfb88, 0xfb9a, 0xfbad, 0xfbbf, 0xfbd1, 0xfbe3, 0xfbf5,
-    0xfc07, 0xfc18, 0xfc2a, 0xfc3b, 0xfc4c, 0xfc5d, 0xfc6e, 0xfc7f, 0xfc8f,
-    0xfca0, 0xfcb0, 0xfcc0, 0xfcd0, 0xfcdf, 0xfcef, 0xfcfe, 0xfd0e, 0xfd1d,
-    0xfd2c, 0xfd3b, 0xfd49, 0xfd58, 0xfd66, 0xfd74, 0xfd83, 0xfd90, 0xfd9e,
-    0xfdac, 0xfdb9, 0xfdc7, 0xfdd4, 0xfde1, 0xfdee, 0xfdfa, 0xfe07, 0xfe13,
-    0xfe1f, 0xfe2b, 0xfe37, 0xfe43, 0xfe4f, 0xfe5a, 0xfe66, 0xfe71, 0xfe7c,
-    0xfe87, 0xfe91, 0xfe9c, 0xfea6, 0xfeb0, 0xfeba, 0xfec4, 0xfece, 0xfed8,
-    0xfee1, 0xfeeb, 0xfef4, 0xfefd, 0xff06, 0xff0e, 0xff17, 0xff1f, 0xff28,
-    0xff30, 0xff38, 0xff3f, 0xff47, 0xff4e, 0xff56, 0xff5d, 0xff64, 0xff6b,
-    0xff71, 0xff78, 0xff7e, 0xff85, 0xff8b, 0xff91, 0xff96, 0xff9c, 0xffa2,
-    0xffa7, 0xffac, 0xffb1, 0xffb6, 0xffbb, 0xffbf, 0xffc4, 0xffc8, 0xffcc,
-    0xffd0, 0xffd4, 0xffd7, 0xffdb, 0xffde, 0xffe1, 0xffe4, 0xffe7, 0xffea,
-    0xffec, 0xffef, 0xfff1, 0xfff3, 0xfff5, 0xfff7, 0xfff8, 0xfffa, 0xfffb,
-    0xfffc, 0xfffd, 0xfffe, 0xffff, 0xffff, 0xffff, 0xffff,
-};
-
 /*
  * angles are measured from -2048 .. 2048
  */
 
 twin_fixed_t twin_sin(twin_angle_t a)
 {
-    twin_fixed_t sin;
-
-    /* limit to [0..360) */
-    a = a & (TWIN_ANGLE_360 - 1);
-    /* special case for 90 degrees - no room in table */
-    if ((a & ~(TWIN_ANGLE_180)) == TWIN_ANGLE_90)
-        sin = TWIN_FIXED_ONE;
-    else {
-        /* mirror second and third quadrant values across y axis */
-        if (a & TWIN_ANGLE_90)
-            a = TWIN_ANGLE_180 - a;
-        sin = _twin_sin_table[a & (TWIN_ANGLE_90 - 1)];
-    }
-    /* mirror third and fourth quadrant values across x axis */
-    if (a & TWIN_ANGLE_180)
-        sin = -sin;
-    return sin;
+    twin_fixed_t sin_val = 0;
+    twin_sincos(a, &sin_val, NULL);
+    return sin_val;
 }
 
 twin_fixed_t twin_cos(twin_angle_t a)
 {
-    return twin_sin(a + TWIN_ANGLE_90);
+    twin_fixed_t cos_val = 0;
+    twin_sincos(a, NULL, &cos_val);
+    return cos_val;
 }
 
 twin_fixed_t twin_tan(twin_angle_t a)
```

Comment:
Is it possible to calculate tan(x) without divisions?

Reply:
I think we could use Taylor series expansion to calculate tan(x) without using division, but as x approaches 90 degrees, the error would become very large. I'm not sure if this is suitable for use in this project.

Reply:
> I think we could use Taylor series expansion to calculate tan(x) without using division, but as x approaches 90 degrees, the error would become very large. I'm not sure if this is suitable for use in this project.

`TWIN_FIXED_MAX` can be set for $tan(\pi/2)$ while `TWIN_FIXED_MIN` can be set for $tan(-\pi/2)$.

Reply:
<img width="798" alt="image" src="https://github.com/user-attachments/assets/77bab338-7d47-4f90-a99b-14cf8afeae91">

Though we could set $tan( \pi /2)$ to `TWIN_FIXED_MAX` directly. However, we still suffer significant error when angle is closed to $\pi/2$. Should we just neglect the error here?

Reply:
@jouae, Can you comment this? Can CORDIC algorithm be applied to tan(x)?

Reply:
> I need to spend some time studying the principles of CORDIC.

See https://github.com/Max-Gulda/Cordic-Math/blob/main/lib/cordicMath/src/cordic-math.c#L290-L332

Reply:
Perhaps we can improve the division about $\frac{x}{y}$

see https://hackmd.io/@sysprog/linux-intdiv

Reply:
> Perhaps we can improve the division about x y
> 
> see https://hackmd.io/@sysprog/linux-intdiv

@jouae Thanks for the follow-up! 
The divisor we use (x calculated in CORDIC or cos calculated in current way) varies for different angles. Is it possible to improve by the mentioned techniques? 
Could you provide a more detailed explanation of your idea? Thanks!

Reply:
@ndsl7109256 
Sure, leave it to me, dude.

Reply:
@jouae, Create a new GitHub issue when this pull request is merged.

Reply:
@jserv 
Roger that.